UnityProject/Assets/Scripts/Objects/Engineering/Reactor/ReactorControlConsole.cs

```diff
@@ -1,14 +1,49 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Text;
+using Communications;
+using InGameEvents;
+using Objects.Machines.ServerMachines.Communications;
 using Shared.Systems.ObjectConnection;
+using Systems.Communications;
+using Systems.Electricity;
 using UnityEngine;
 
-namespace Objects.Engineering
+namespace Objects.Engineering.Reactor
 {
-	public class ReactorControlConsole : MonoBehaviour, IMultitoolSlaveable
+	[RequireComponent(typeof(APCPoweredDevice))]
+	[RequireComponent(typeof(Integrity))]
+	public class ReactorControlConsole : SignalEmitter, IMultitoolSlaveable, IChatInfluencer
 	{
 		[SceneObjectReference] public ReactorGraphiteChamber ReactorChambers = null;
 
-		[SceneObjectReference] public List<ReactorGraphiteChamber> ReactorChambers2 = new List<ReactorGraphiteChamber>();
+		private APCPoweredDevice poweredDevice;
+		private Integrity integrity;
+		private UniversalObjectPhysics objectPhysics;
+
+		private ChatEvent chatEvent = new ChatEvent();
+		private const ChatChannel ChatChannels = ChatChannel.Engineering | ChatChannel.Common;
```

Comment:
this possibly should only broadcast to Engineering?

Reply:
why?

Reply:
that's how the SM behaves as well, isn't it?

Reply:
uhh, i don't recall

Reply:
it only talks to engineering now